common/report.c

```diff
@@ -54,9 +54,12 @@ static oe_result_t _oe_parse_sgx_report_body(
             report_body->mrsigner,
             sizeof(report_body->mrsigner)));
 
-    parsed_report->identity.product_id[0] = report_body->isvprodid & 0xFF;
+    if (report_body->isvprodid > OE_INT8_MAX)
+        goto done;
+    parsed_report->identity.product_id[0] =
+        (uint8_t)report_body->isvprodid & 0xFF;
     parsed_report->identity.product_id[1] =
-        (report_body->isvprodid >> 8) & 0xFF;
+        ((uint8_t)report_body->isvprodid >> 8) & 0xFF;
```

Comment:
This line introduces a bug.  The cast operator has higher precedence than the right shift operator. Casting down to a uint8_t before shifting right by 8 bits causes this expression to always evaluate to zero

Reply:
Suggest changing to:
```
(uint8_t)((report_body->isvprodid >> 8) & 0xFF);
```

This might be a little overkill, but I like being as explicit as possible with casts, since sometimes operation order in a complicated expression is sometimes not obvious at all